src/components/Experience.ts

```diff
@@ -15,9 +16,15 @@ export class Experience<TGameStartr extends FullScreenPokemon> extends Component
     public levelup(pokemon: IPokemon): void {
         pokemon.level += 1;
         pokemon.statistics = this.gameStarter.equations.newPokemonStatistics(pokemon.title, pokemon.level, pokemon.ev, pokemon.iv);
-        
+
         // TODO: display text box containing levelup info
-        // TODO: add in evolution check/evolve the pokemon here
+        
+        const evo: Evolution<TGameStartr> = new Evolution(this.gameStarter);
```

Comment:
As you suggested and we discussed offline: this should be elsewhere, and only instantiated once.